src/AppKit/NSView.cs

```diff
@@ -31,20 +33,25 @@ sealed class SortData {
 #endif
 		static nint view_compare (IntPtr view1, IntPtr view2, IntPtr context)
 		{
-			var data = (SortData) GCHandle.FromIntPtr (context).Target;
+			var data = GCHandle.FromIntPtr (context).Target as SortData;
 			try {
-				var a = (NSView) Runtime.GetNSObject (view1);
-				var b = (NSView) Runtime.GetNSObject (view2);
-				return (nint) (long) data.Comparer (a, b);
+				if (Runtime.GetNSObject (view1) is NSView a
+						&& Runtime.GetNSObject (view2) is NSView b) {
+					var result = data?.Comparer?.Invoke (a, b);
+					return result is null ? (nint) (long) NSComparisonResult.Same : (nint) (long) result;
+				} else {
+					return (nint) (long) NSComparisonResult.Same;
+				}
```

Comment:
You can mostly use the old code if you change the comparison function to allow null items (see further below):

```suggestion
				var a = Runtime.GetNSObject (view1) as NSView;
				var b = Runtime.GetNSObject (view2) as NSView;
				return (nint) (long) data.Comparer (a, b);
```